.github/workflows/ompi_mpi4py.yaml

```diff
@@ -118,3 +167,135 @@ jobs:
       run:  python demo/test-run/test_run.py -v
       if:   ${{ true }}
       timeout-minutes: 10
+
+  #==============================================
+
+  run_spawn:
```

Comment:
```suggestion
  run_spawn:
    if: ${{ github.event.label.name == 'mpi4py-spawn' }}
```

Reply:
This is perfect; thanks!

Reply:
It looks like `github.event.label` is the event fired when you set a label -- it doesn't check whether a label is set on the PR itself.  This Github Action doesn't fire on the `event` label.  I'll investigate more in a bit...

Reply:
@dalcinl Thanks for all that; that was most helpful in understanding what to do.

I found the proper `if` conditional:

```
if: ${{ contains(github.event.pull_request.labels.*.name, 'mpi4py-all') }}
```

This looks through all the labels on the PR right now -- not just for the labeling event -- and sees if that label is there.  A side effect here is that you have to set the label and then force the github action to run again (e.g., via a meaningless commit / push), but I'm ok with that.

Reply:
> (e.g., via a meaningless commit / push), but I'm ok with that.

You should be able to manually re-run a workflow using the web UI, look at the top right. I'm not sure if this would actually re-evaluate label changes, but worth a try...

![image](https://github.com/open-mpi/ompi/assets/294981/7832a69b-b29f-428a-95f2-a9770517c84d)

Reply:
I did try this over the weekend; it seems to send the same github context blob to the new run that it sent to the original run, so the label status is the same.

I'm ok with the PR author needing to push a new commit to the PR to get it to trigger the spawn/dynamic tests if they didn't originally label the PR that way.